File: backend/src/main/java/com/nhtl/dto/auth/SignupResponse.java
package com.nhtl.dto.auth;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class SignupResponse {
    
    private String userId;  // ✅ STRING, pas UUID!
    private String email;
    private String phone;
    private String username;
    private String message;
}

File: frontend/lib/models/auth/signup_response.dart
class SignupResponse {
  final String userId;
  final String email;
  final String phone;
  final String username;
  final String message;

  SignupResponse({
    required this.userId,
    required this.email,
    required this.phone,
    required this.username,
    required this.message,
  });

  factory SignupResponse.fromJson(Map<String, dynamic> json) {
    return SignupResponse(
      userId: json['userId'] ?? '',
      email: json['email'] ?? '',
      phone: json['phone'] ?? '',
      username: json['username'] ?? '',
      message: json['message'] ?? '',
    );
  }
}
